cfme/networks/provider/nuage.py

```diff
@@ -155,3 +156,99 @@ def num_floating_ip(self):
+        return view.entities.properties.get_text_of('Type')
+
+    @variable(alias="ui")
+    def cidr_value(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of('CIDR')
+
+    @variable(alias="ui")
+    def gateway_value(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of('Gateway')
+
+    @variable(alias="ui")
+    def network_protocol_value(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of('Network protocol')
+
+    @variable(alias="ui")
+    def network_manager_value(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.relationships.get_text_of('Network Manager')
+
+    @variable(alias="ui")
+    def cloud_tenant_value(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.relationships.get_text_of('Cloud tenant')
+
+    @variable(alias="ui")
+    def network_router_value(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.relationships.get_text_of('Network Router')
+
+    @variable(alias="ui")
+    def network_ports_num(self):
+        view = navigate_to(self, 'Details')
+        return int(view.entities.relationships.get_text_of('Network Ports'))
+
+    @variable(alias="ui")
+    def security_groups_num(self):
+        view = navigate_to(self, 'Details')
+        return int(view.entities.relationships.get_text_of('Security Groups'))
+
+    def validate_stats(self, host_stats, ui=False):
+        """ Validates that the detail page matches the subnet's inventory.
+
+        Args:
+            host_stats: dictionary of values to be compered to UI values,
+            where keys have the same names as NuageSubnet ui functions
+
+        This method gets host stats and matches them against the UI. An exception will
+        be raised if the stats retrieved from the UI do not match those from host stats.
+        """
+
+        # Make sure we are on the subnet detail page and refresh it
+        if ui:
+            navigate_to(self, "Details")
+            self.browser.selenium.refresh()
+
+        cfme_stats = {stat: getattr(self, stat)(method='ui' if ui else None) for stat in host_stats}
+        assert cfme_stats == host_stats
+
+
+@attr.s
+class NuageSubnetCollection(SubnetCollection):
+    ENTITY = NuageSubnet
+
+    def find_by_ems_ref(self, ems_ref):
+        """returning NuageSubnet object with provided ems_ref"""
+        subnets_table = self.appliance.db.client['cloud_subnets']
+        subnet = None
+        try:
+            subnet_query = (
+                self.appliance.db.client.session
+                    .query(subnets_table.name, subnets_table.ems_ref)
+                    .filter(subnets_table.ems_ref == ems_ref)).all()[0]
+            subnet = self.instantiate(name=subnet_query.name, ems_ref=subnet_query.ems_ref)
+        finally:
+            return subnet
```

Comment:
Nah, there are multiple things that bother me: first, we suppress all the errors even potential syntax errors with try-finally. Second, we call `.all()[0]` on query instead `.first()`. Third, we the docstring is not capitalized. Can I suggest something like:

```python
def find_by_ems_ref(self, ems_ref):
        """Fetch NuageSubnet instance with provided ems_ref"""
        subnets_table = self.appliance.db.client['cloud_subnets']
        subnet = self.appliance.db.client.session.query(subnets_table.name, subnets_table.ems_ref).filter(subnets_table.ems_ref == ems_ref).first()

        return self.instantiate(name=subnet.name, ems_ref=subnet.ems_ref) if subnet else None
```

Btw, the ORM that is used here is sqlalchemy, which comes with as powerful ORM as the one shipped with Django. Here's the doc on [.first()](http://docs.sqlalchemy.org/en/rel_0_9/orm/query.html#sqlalchemy.orm.query.Query.first) method. It returns `None` if no records are found so no need for try-catch anymore.